src/libbson/src/jsonsl/jsonsl.c

```diff
@@ -1149,6 +1149,9 @@ void jsonsl_jpr_match_state_init(jsonsl_t jsn,
         return;
     }
     jsn->jprs = (jsonsl_jpr_t *)malloc(sizeof(jsonsl_jpr_t) * njprs);
+    if (!jsn->jprs) {
+        return;
+    }
     jsn->jpr_count = njprs;
     jsn->jpr_root = (size_t*)calloc(1, sizeof(size_t) * njprs * jsn->levels_max);
```

Comment:
This `calloc` could also fail (in which cast the `malloc` above would need to be free'd as well)

Reply:
Good catch. I missed that. Commit incoming.